test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -538,6 +539,12 @@ class test_uct_cm_sockaddr : public uct_test {
                      (TEST_CM_STATE_SERVER_CONNECTED  | TEST_CM_STATE_CLIENT_CONNECTED |
                       TEST_CM_STATE_CLIENT_GOT_REJECT | TEST_CM_STATE_CLIENT_GOT_ERROR));
 
+        deadline = ucs_get_time() + ucs_time_from_sec(DEFAULT_TIMEOUT_SEC) *
+                                    ucs::test_time_multiplier();
+
+        while ((m_delayed_conn_reqs.size() == 0) && (ucs_get_time() < deadline)) {
```

Comment:
can be simplified just moving 
```
        self->m_cm_state |= TEST_CM_STATE_CONNECT_REQUESTED;
```
to the end of `cm_conn_request_cb`
std::vector is not thread safe

Reply:
moving this flag to the end of the cb will mean to set CONNECT_REQUESTED after the server already accepted and therefore may already be connected.

Reply:
hmm... ok, then need to handle it differently in branches
```
        if (self->m_delay_conn_reply) {
            self->m_delayed_conn_reqs.push(conn_request);
        } else if (self->m_reject_conn_request) {
            status = uct_listener_reject(listener, conn_request);
            ASSERT_UCS_OK(status);
            self->m_cm_state |= TEST_CM_STATE_SERVER_REJECTED;
        } else {
            self->server_accept(self->m_server, conn_request,
                                server_connect_cb, server_disconnect_cb, self);
        }
```
or add a lock (can reuse async context for CM tests), using `m_delayed_conn_reqs.size()` is wrong because it's not thread safe

Reply:
do you see a case where a race is possible here? there is one client and a request is added to the  queue only in once place.

Reply:
potential race between `push_back` and `size` which depends on `std::vector` implementation